### PR TITLE
Use a queue and separate task to process BLE devices.

### PR DESCRIPTION
**Pull request by h2zero**
## Description:
This avoids excess processing in the BLE advertised device callback and prevents overflowing the BLE host stack.

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by h2zero** [comment]
@NorthernMan54 Could you test this? It should resolve the stack issue you mentioned without altering the BLE stack size.

**Comment by NorthernMan54** [comment]
Still getting the canary issue

```
T: [ OMG->MQTT ] topic: home/240AC4EC20DC/SYStoMQTT msg: {"uptime":10,"version":"ddd-bt-rtl-v0.9.8-38-g4dccdb3[pr/1110]","freemem":142064,"mqttport":"1883","mqttsecure":false,"freestack":4012,"rssi":-49,"SSID":"67 Bonacres","ip":"192.168.1.181","mac":"24:0A:C4:EC:20:DC","lowpowermode":0,"btqblck":0,"btqsum":0,"btqsnd":0,"btqavg":0,"interval":55555,"scanbcnct":10,"scnct":0,"modules":["BT"]} 
N: Scan begin
T: Creating BLE buffer
N: Device detected: 61:EA:8C:4F:88:06
T: getDeviceByMac 61:EA:8C:4F:88:06
Guru Meditation Error: Core  1 panic'ed (Unhandled debug exception)
Debug exception reason: Stack canary watchpoint triggered (procBLETask) 
Core 1 register dump:
PC      : 0x40093ea7  PS      : 0x00060736  A0      : 0x4009387e  A1      : 0x3ffe0410  
A2      : 0x3ffe0480  A3      : 0x3ff94f64  A4      : 0x00000000  A5      : 0x0000005b  
A6      : 0x00000040  A7      : 0x00000009  A8      : 0x800d738d  A9      : 0x4009396c  
A10     : 0x00000001  A11     : 0x3ffe514b  A12     : 0x4008e108  A13     : 0x00000001  
A14     : 0x78006633  A15     : 0x37303530  SAR     : 0x00000011  EXCCAUSE: 0x00000001  
EXCVADDR: 0x00000000  LBEG    : 0x400012c5  LEND    : 0x400012d5  LCOUNT  : 0xfffffffa  

ELF file SHA256: 0000000000000000

Backtrace: 0x40093ea7:0x3ffe0410 0x4009387b:0x00000000
  #0  0x40093ea7:0x3ffe0410 in xthal_save_extra_nw at /Users/igrokhotkov/e/esp32/hal/hal/state_asm.S:72
  #1  0x4009387b:0x00000000 in _xt_context_save at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/xtensa_context.S:194

Rebooting...
```

**Comment by h2zero** [comment]
Thanks, looks like its in the new task now instead of the BLE stack task. This can be bumped up and monitored now.

File: main/ZgatewayBT.ino [hunk 1]
```diff
@@ -585,14 +604,25 @@ void setupBT() {
...
 
+  BLEQueue = xQueueCreate(32, sizeof(NimBLEAdvertisedDevice*));
+
   BLEDevice::setScanDuplicateCacheSize(BLEScanDuplicateCacheSize);
   BLEDevice::init("");
 
+  xTaskCreatePinnedToCore(
+      procBLETask, /* Function to implement the task */
+      "procBLETask", /* Name of the task */
+      4096, /* Stack size in bytes */
```
Comment (h2zero) [inline main/ZgatewayBT.ino]: We'll need to find a sufficient size for this. Setting log level to trace will print the high water mark for this stack.

**Comment by NorthernMan54** [comment]
I'm go to let this run for a bit to watch for any leaks etc

To get it to work I needed to configure the stack size as 8096, and it is reporting free currently as 3596.

**Comment by h2zero** [comment]
Great, thanks! 5120 might be the magic number, more testing needed though.

**Comment by NorthernMan54** [comment]
Ran for a couple of hours and stack was stable

T: Process BLE stack free: 620

Looks like you nailed it

**Comment by h2zero** [comment]
Thanks, I appreciate the help. I'll update the value to 5120.

**Comment by 1technophile** [comment]
Thanks both of you!

**Pull request merged**